src/java/apiview-java-processor/src/main/java/com/azure/tools/apiview/processor/analysers/util/ASTUtils.java

```diff
@@ -209,7 +207,34 @@ public static String makeId(EnumConstantDeclaration enumDeclaration) {
     }
 
     public static String makeId(String fullPath) {
-        return MAKE_ID.matcher(fullPath).replaceAll("-");
+        if (fullPath == null || fullPath.isEmpty()) {
+            return fullPath;
+        }
+
+        StringBuilder sb = null;
+        int prevStart = 0;
+
+        int length = fullPath.length();
+        for (int i = 0; i < length; i++) {
+            char c = fullPath.charAt(i);
+            if (c == '"' || c == ' ') {
+                if (sb == null) {
+                    sb = new StringBuilder(length);
+                }
+
+                if (prevStart != i) {
+                    sb.append(fullPath, prevStart, i);
+                }
```

Comment:
Should we also append `-` after this to keep the ids backward compatible?

Reply:
Good catch, missed that (more likely I had it then reverted a change and forgot it again)